### PR TITLE
feat(cli): add check for migrations command

### PR DESCRIPTION
**Pull request by kpervin**
Allow checking for migrations. This would be useful for adding to git hooks to ensure that any changes to ORM entities are captured in migrations.

### CODE REVIEW

File: packages/cli/src/commands/MigrationCommandFactory.ts [hunk 1]
```diff
@@ -164,6 +168,16 @@ export class MigrationCommandFactory {
     CLIHelper.dump(colors.green(`${ret.fileName} successfully created`));
   }
 
+  private static async handleCheckCommand(migrator: IMigrator, orm: MikroORM): Promise<void> {
+    const diff = await migrator.checkMigrationNeeded();
```
Comment (B4nan) [inline packages/cli/src/commands/MigrationCommandFactory.ts:172]: that method should return boolean i guess

File: packages/cli/src/commands/MigrationCommandFactory.ts [hunk 2]
```diff
@@ -164,6 +168,16 @@ export class MigrationCommandFactory {
...
+
+    if (diff.up.length === 0) {
+      return CLIHelper.dump(colors.green(`No changes required, schema is up-to-date`));
+    }
+    await orm.close(true);
+    throw new Error(`Changes detected. Please create migrations to update schema.`);
```
Comment (B4nan) [inline packages/cli/src/commands/MigrationCommandFactory.ts:178]: i am not entirely sure i like the idea of throwing here. maybe `process.exit(1)` would be better

Comment (kpervin) [reply packages/cli/src/commands/MigrationCommandFactory.ts:178]: That's funny, because originally I was going to use `process.exit(1)` but thought it might be better to throw an error lol

Comment (B4nan) [reply packages/cli/src/commands/MigrationCommandFactory.ts:178]: i dont like the idea of showing a stack trace for this, sounds quite confusing. and for automation you only care about the exit code

File: packages/cli/src/commands/MigrationCommandFactory.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ export class MigrationCommandFactory {
 
   static readonly DESCRIPTIONS = {
     create: 'Create new migration with current schema diff',
+    check: 'Check if migrations are needed. Useful for bash scripts.',
```
Comment (B4nan) [inline packages/cli/src/commands/MigrationCommandFactory.ts:11]: lets move this to less prominent place, maybe after `list` command

File: docs/docs/migrations.md [hunk 4]
```diff
@@ -161,6 +161,7 @@ You can use it via CLI:
 
 ```sh
 npx mikro-orm migration:create   # Create new migration with current schema diff
+npx mikro-orm migration:check    # Check if schema is up to date
```
Comment (B4nan) [inline docs/docs/migrations.md:164]: lets move this to less prominent place, maybe after list command

File: packages/cli/src/commands/MigrationCommandFactory.ts [hunk 5]
```diff
@@ -164,6 +168,15 @@ export class MigrationCommandFactory {
     CLIHelper.dump(colors.green(`${ret.fileName} successfully created`));
   }
 
+  private static async handleCheckCommand(migrator: IMigrator, orm: MikroORM): Promise<void> {
+    if (!await migrator.checkMigrationNeeded()) {
+      return CLIHelper.dump(colors.green(`No changes required, schema is up-to-date`));
+    }
+    await orm.close(true);
+    CLIHelper.dump(colors.yellow(`Changes detected. Please create migrations to update schema.`));
```
Comment (B4nan) [inline packages/cli/src/commands/MigrationCommandFactory.ts:176]:
```suggestion
    CLIHelper.dump(colors.yellow(`Changes detected. Please create migration to update schema.`));
```

File: packages/migrations/src/Migrator.ts [hunk 6]
```diff
@@ -51,7 +51,6 @@ export class Migrator implements IMigrator {
     if (initial) {
       return this.createInitialMigration(path);
     }
-
```
Comment (B4nan) [inline packages/migrations/src/Migrator.ts]: unrelated change, the line break was there on purpose

**Review by B4nan — COMMENTED** [review]
looking good, just few nits, thanks!

**Comment by B4nan** [comment]
and please dont rebase after you get a review, i cant see what you changed. will be squash merging the PR so it does not really matter

**Comment by kpervin** [comment]
>

**Review by B4nan — APPROVED** [review]
